utils/frame/try-runtime/cli/src/lib.rs

```diff
@@ -60,31 +56,93 @@ pub struct TryRuntimeCmd {
 		default_value = "Interpreted"
 	)]
 	pub wasm_method: WasmExecutionMethod,
+
+	/// The state to use to run the migration.
+	#[structopt(subcommand)]
+	pub state: State,
 }
 
 /// The state to use for a migration dry-run.
-#[derive(Debug)]
+#[derive(Debug, structopt::StructOpt)]
 pub enum State {
-	/// A snapshot. Inner value is a file path.
-	Snap(String),
+	/// Use a snapshot as state to run the migration.
+	Snap {
+		#[structopt(flatten)]
+		cache_params: CacheParams,
+	},
+
+	/// Use a live chain to run the migration.
+	Live {
+		/// An optional cache file to WRITE to. Not cached if set to `None`.
+		#[structopt(short, long)]
+		cache_file: Option<CacheParams>,
+
+		/// The block hash at which to connect.
```

Comment:
you support number as well, doc is outdated.

Reply:
I removed number support, because I had some problems with it. So we're back to only hashes.

Reply:
ahh okay. I just realized that as well.